elementary/monitor/data_monitoring/data_monitoring_alerts.py

```diff
@@ -66,31 +74,89 @@ def _get_user_id(email: str) -> str:
         else:
             return prettify_json_str_set(emails)
 
-    def _send_alerts_to_slack(self, alerts: List[Alert], alerts_table_name: str):
-        if not alerts:
-            return
-
-        sent_alert_ids = []
-        alerts_with_progress_bar = alive_it(alerts, title="Sending alerts")
-        for alert in alerts_with_progress_bar:
+    def _fix_owners_and_subscribers(self, group_alert: GroupOfAlerts):
+        """
+        Only reason it's not in the __init__ of GroupOfAlerts, is that it goes to the slack API.
+        This function is based on _parse_emails_to_ids which uses slack's API to get the handle for owners, subscribers.
+        :param group_alert:
+        :return:
+        """
+        for alert in group_alert.alerts:
             alert.owners = self._parse_emails_to_ids(alert.owners)
             alert.subscribers = self._parse_emails_to_ids(alert.subscribers)
-            alert_msg = alert.to_slack()
-            sent_successfully = self.slack_client.send_message(
-                channel_name=alert.slack_channel
-                if alert.slack_channel
-                else self.config.slack_channel_name,
-                message=alert_msg,
-            )
-            if sent_successfully:
-                sent_alert_ids.append(alert.id)
-            else:
+        all_owners = set([])
+        all_subscribers = set([])
+        for alert in group_alert.alerts:
+            all_owners.update(alert.owners)
+            all_subscribers.update(alert.subscribers)
+        group_alert.owners = all_owners
+        group_alert.subscribers = all_subscribers
+
+    def _group_alerts_per_config(self, alerts: List[Alert]) -> List[GroupOfAlerts]:
+        """
+        reads self.config and alerts' config, and groups alerts in a smart way
+        1. split by grouping type
+        2. split Table grouped-by, by the Table
+        3. concat
+
+        :param alerts:
+        :return:
+        """
+        # import pdb; pdb.set_trace()
+        default_alerts_group_by_strategy = GroupingType(
+            self.config.slack_group_alerts_by
+        )
+        alerts_by_grouping_mechanism = defaultdict(lambda: [])
+        for al in alerts:
+            if not al.slack_group_alerts_by:
+                alerts_by_grouping_mechanism[default_alerts_group_by_strategy].append(
+                    al
+                )
+                continue
+            try:
+                t = GroupingType(al.slack_group_alerts_by)
```

Comment:
please use type / grouping_type instead of t

Reply:
Agreed

Reply:
accept

Reply:
done